src/shape.rs

```diff
@@ -348,3 +376,356 @@ mod test
+    }
+
+    /// Returns a vector representing the bottom left vertex.
+    #[inline]
+    pub fn bottom_left(&self) -> Vector2<T>
+    {
+        Vector2::new(self.top_left.x, self.bottom_right.y)
+    }
+
+    /// Returns the radius of the rounded corners.
+    #[inline]
+    pub fn radius(&self) -> T
+    {
+        self.radius
+    }
+
+    /// Returns the x value of the left border
+    #[inline]
+    pub fn left(&self) -> T
+    {
+        self.top_left.x
+    }
+
+    /// Returns the x value of the right border
+    #[inline]
+    pub fn right(&self) -> T
+    {
+        self.bottom_right.x
+    }
+
+    /// Returns the y value of the top border
+    #[inline]
+    pub fn top(&self) -> T
+    {
+        self.top_left.y
+    }
+
+    /// Returns the y value of the bottom border
+    #[inline]
+    pub fn bottom(&self) -> T
+    {
+        self.bottom_right.y
+    }
+
+    /// Returns a `Rectangle` representing the rectangle that encloses this
+    /// rounded rectangle.
+    #[inline]
+    pub fn as_rectangle(&self) -> Rectangle<T>
+    {
+        Rectangle::new(self.top_left, self.bottom_right)
+    }
+}
+
+impl<T: std::ops::Sub<Output = T> + Copy> RoundedRectangle<T>
+{
+    /// Returns the width of the rounded rectangle.
+    #[inline]
+    pub fn width(&self) -> T
+    {
+        self.bottom_right.x - self.top_left.x
+    }
+
+    /// Returns the height of the rounded rectangle.
+    #[inline]
+    pub fn height(&self) -> T
+    {
+        self.bottom_right.y - self.top_left.y
+    }
+
+    /// Returns a `Vector2` containing the width and height of the rounded
+    /// rectangle.
+    #[inline]
+    pub fn size(&self) -> Vector2<T>
+    {
+        Vector2::new(self.width(), self.height())
+    }
+}
+
+impl<T> RoundedRectangle<T>
+where
+    T: num_traits::AsPrimitive<f32>
+        + std::cmp::PartialOrd
+        + std::ops::Add<Output = T>
+        + std::ops::Sub<Output = T>
+        + std::ops::Mul<Output = T>
+        + std::ops::Neg<Output = T>
+        + std::ops::Div<Output = f32>
+        + std::ops::Div<f32, Output = T>
+{
+    /// Returns true if the specified point is inside this rounded rectangle.
+    /// Note: this is always inclusive, in contrast to the `contains` method
+    /// of `Rect` which is sometimes exclusive.
+    #[inline]
+    #[must_use]
+    pub fn contains(&self, point: Vector2<T>) -> bool
+    {
+        //if outside the enclosing rectangle then call it a win and don't proceed
+        if point.x <= self.left()
+            || point.x >= self.right()
+            || point.y >= self.bottom()
+            || point.y <= self.top()
+        {
+            return false;
+        }
+
+        //...by looking at the rounded rectangle as 2 rectangles in a cross and 4
+        //...by circles (overlapping rectangles should be slightly better
+        // than 3 rectangles in this case (I think)):
+        //first rectangle:
+        if point.x >= self.top_left.x
```

Comment:
This is good, although I think it would be cleaner to make a new local `Rectangle` for these two cases, so that we can just call `contains()` (rather than listing out the eight comparisons)